### PR TITLE
Conversion of Cloud API Demo bash Scripts to Python 

### LINKED ISSUE #1428: Cloud Based API Instance for Developers
We need to have a cloud based API instance for developers to test against. It must:

1. Be integrated into our CI/CD pipeline so that the app is rebuilt with every newly merged PR.
1. Initialize the database to a known working state with appropriate services will be restarted when this happens: (there are existing API calls for this):
    1. with each CI/CD update
    1. every 24 hours
1. Use a talawa.io sub-domain as the endpoint
1. The API instance will have a rate limiting mechanism to reduce the risk of abuse 
1. Preferably use a free service
1. Use a palisadoes role based account for management
1. The demo cloud API must be easily switched between branches as the code source. We will be migrating to master as the default branch soon.

Please coordinate these efforts with @noman2002 and @kb-0311. Ask to be assigned this task by them.

### PR DESCRIPTION
**Pull request by vasujain275**
**What kind of change does this PR introduce?**

This pr refactor the scripts present in `scripts/cloud-api-demo` from bash to python, this is related to https://github.com/PalisadoesFoundation/talawa-api/issues/1428 but do not completely fixes the issue.

**Issue Number:**

Fixes #1428

**Did you add tests for your changes?**

No

**Snapshots/Videos:**

N/A

**If relevant, did you update the documentation?**

No

**Summary**

This pr converts all the important scripts needed to function talawa api's cloud instance to python. It follows all the guidelines given here - https://github.com/PalisadoesFoundation/talawa-api/issues/1428#issuecomment-1890588216

**Does this PR introduce a breaking change?**

No

**Other information**

To test these scripts on demo godaddy vps, log in as `talawa-api` user and use the following commands:

```bash
python3 deploy.py --path ~/develop/talawa-api --branch develop
```

```bash
python3 renew_certificates.py --config-dir ~/.certbot/config --logs-dir ~/.certbot/logs --work-dir ~/.certbot/work
```

```bash
python3 reset_database.py --mongo-container talawa-api_mongodb_1 --mongo-db talawa-api --repo-dir /home/talawa-api/develop/talawa-api
```

```bash
python3 correct_permissions.py --user talawa-api --files /usr/local/bin/deploy.py /usr/local/bin/reset_database.py /usr/local/bin/renew_certificates.py
```

**I have tested all the scripts on the demo vps and they are working perfectly.**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. Make the code match the Google style guide
    1. https://google.github.io/styleguide/pyguide.html#383-functions-and-methods
    image
1. Make sure the code is pylint, python black, flake8, and pydocstyle compliant. These are pip lint and formatting packages that you can use in your IDE

File: scripts/cloud-api-demo/correct_permissions.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import argparse
+import subprocess
+
+def execute_command(cmd):
+    """Execute a shell command."""
+    process = subprocess.run(cmd, capture_output=True, text=True)
+
+    if process.returncode != 0:
+        print(f"Error occurred: {process.stderr}")
```
Comment (palisadoes) [inline scripts/cloud-api-demo/correct_permissions.py:9]: If there is an error it must immediately exit with a non zero exit code after printing the message

File: scripts/cloud-api-demo/deploy.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import subprocess
+import os
+
+def execute_command(cmd):
+    # Execute a shell command.
+    process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    output, error = process.communicate()
+
+    if process.returncode != 0:
+        print(f"Error occurred: {error.decode().strip()}")
```
Comment (palisadoes) [inline scripts/cloud-api-demo/deploy.py:11]: If there is an error it must immediately exit with a non zero exit code after printing the message

File: scripts/cloud-api-demo/renew_certificates.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import argparse
+import subprocess
+
+def execute_command(cmd):
+    """Execute a shell command."""
+    process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    output, error = process.communicate()
+
+    if process.returncode != 0:
+        print(f"Error occurred: {error.decode().strip()}")
```
Comment (palisadoes) [inline scripts/cloud-api-demo/renew_certificates.py:10]: If there is an error it must immediately exit with a non zero exit code after printing the message

File: scripts/cloud-api-demo/correct_permissions.py (hunk 1 above)
Comment (vasujain275) [reply scripts/cloud-api-demo/correct_permissions.py:17]: Implemented

File: scripts/cloud-api-demo/deploy.py (hunk 2 above)
Comment (vasujain275) [reply scripts/cloud-api-demo/deploy.py:19]: Implemented

File: scripts/cloud-api-demo/renew_certificates.py (hunk 3 above)
Comment (vasujain275) [reply scripts/cloud-api-demo/renew_certificates.py:18]: Implemented

**Comment by vasujain275** [comment]
@palisadoes 

Thank you for the review, I have added the changes requested that are - 
1. Linted the python files properly.
2. Added `raise SystemExit(1)` to exit with a non zero exit code after printing the error message.

File: scripts/cloud-api-demo/correct_permissions.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import argparse
```
Comment (palisadoes) [inline scripts/cloud-api-demo/correct_permissions.py:1]: This must be the first line to specify the specific interpreter to use

```
#!/usr/bin/env python3
```

File: scripts/cloud-api-demo/renew_certificates.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import argparse
```
Comment (palisadoes) [inline scripts/cloud-api-demo/renew_certificates.py:1]: This must be the first line to specify the specific interpreter to use

```
#!/usr/bin/env python3
```

File: scripts/cloud-api-demo/reset_database.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
+import argparse
```
Comment (palisadoes) [inline scripts/cloud-api-demo/reset_database.py:1]: This must be the first line to specify the specific interpreter to use

```
#!/usr/bin/env python3
```

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments

File: scripts/cloud-api-demo/correct_permissions.py (hunk 4 above)
Comment (vasujain275) [reply scripts/cloud-api-demo/correct_permissions.py:2]: Added

File: scripts/cloud-api-demo/renew_certificates.py (hunk 5 above)
Comment (vasujain275) [reply scripts/cloud-api-demo/renew_certificates.py:2]: Added

File: scripts/cloud-api-demo/reset_database.py (hunk 6 above)
Comment (vasujain275) [reply scripts/cloud-api-demo/reset_database.py:2]: Added

**Review by palisadoes — APPROVED** [review]

**Pull request merged**